### PR TITLE
Filter Select children

### PR DESCRIPTION
**Pull request by Joozty**
Allows to use this kind of syntax:

```
const booleanValue = false
<Select>
  {booleanValue && <Select.Option />}
</Select>
```

### CODE REVIEW

File: src/modules/select/hooks/use-options.js [hunk 1]
```diff
@@ -22,34 +22,39 @@ const useOptions = (
...
-        },
-        'onMouseUp': (event) => {
-          if (clickedElement === element.props.id) {
-            changeActiveOption({
-              value: element.props.value,
-              id: element.props.id,
-            })
-          }
+    React.Children.toArray(children)
+      .filter((child) => child.type.displayName === 'SelectOption')
```
Comment (viktor-yakubiv) [inline src/modules/select/hooks/use-options.js:26]: What if the `child` is `null`? Is it filtered out already by `React.Children.toArray`?

File: src/modules/select/hooks/use-options.js [hunk 2]
```diff
@@ -22,34 +22,39 @@ const useOptions = (
...
+        const overrideProps = {
+          'id': `suggestion-result-${selectId}-${element.props.id}`,
+          'value': element.props.value,
+          'data-select-id': element.props.id,
+          'onMouseDown': () => {
+            // Firefox and Safari have relatedTarget null.
+            // Therefore we cannot use onClick event
+            // since it's not getting fired
+            // because onBlur input listener cause rerender.
+            setClickedElement(element.props.id)
```
Comment (viktor-yakubiv) [inline src/modules/select/hooks/use-options.js:37]: Is it actually clicked, hovered, focused or selected?

**Review by viktor-yakubiv — APPROVED** [review]
It's a good improvement!

File: src/modules/select/hooks/use-options.js (hunk 1 above)
Comment (Joozty) [reply src/modules/select/hooks/use-options.js:26]: Looks like it is indeed

File: src/modules/select/hooks/use-options.js (hunk 2 above)
Comment (Joozty) [reply src/modules/select/hooks/use-options.js:37]: It's an element which received onMouseDown event. If the same element receives onMouseUp event then it is considered to be clicked.

**Pull request merged**